docs/api/StreamStart.md

```diff
@@ -17,15 +17,40 @@ QUIC_STATUS
 
 # Parameters
 
-**TODO**
+`Stream`
+
+The valid handle to an open stream object.
+
+`Flags`
+
+The set of flags the controls the behavior of `StreamStart`:
+
+Value | Meaning
+--- | ---
+**QUIC_STREAM_START_FLAG_NONE**<br>0 | No special behavior. Executes as a blocking call, returning only after the operation has completed. The peer is not informed of the stream starting until the app sends data.
+**QUIC_STREAM_START_FLAG_FAIL_BLOCKED**<br>1 | Only opens the stream if flow control (from the peer) allows. If the stream count limit has currently been reached the start will fail with `QUIC_STATUS_STREAM_LIMIT_REACHED`.
+**QUIC_STREAM_START_FLAG_IMMEDIATE**<br>2 | Indicates that the peer should be immediately informed of the stream opening (or at least as soon as flow control allows) and not wait for any data to be sent first.
+**QUIC_STREAM_START_FLAG_ASYNC**<br>4 | Causes the API to be non-blocking. The API call itself will just queue the operation and immediate return. The final result of the start operation will be delivered via the `QUIC_STREAM_EVENT_START_COMPLETE` event.
+**QUIC_STREAM_START_FLAG_SHUTDOWN_ON_FAIL**<br>8 | Causes the stream to immediate shutdown (abortive) if the start operation fails.
+**QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT**<br>16 | Indicates the app wants the `QUIC_STREAM_EVENT_PEER_ACCEPTED` event to be delivered if the stream isn't initially accepted (allowed by flow control) at start completion.
```

Comment:
Not clear to me when this event would happen -- immediately after sending the first frame for the stream, or after it is acked?

Reply:
Also -- when this operation blocks due to flow control, can it be cancelled? How? This is something we will likely need for the managed API.

Reply:
You can still call shutdown on the stream (or the connection). If you shutdown the stream, flow control will also prevent the operation from being sent over the wire. But you can abort and use the immediate flags to get the shutdown complete event and then close the stream and be done with it as far as the app is concerned. MsQuic will maintain it queued up to indicate the peer if/when it can.